nixos/modules/services/databases/terminusdb.nix

```diff
@@ -0,0 +1,63 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  format = pkgs.formats.json { };
+  cfg = config.services.terminusdb;
+in
+{
+  options = {
+    services.terminusdb = {
+      enable = mkEnableOption (lib.mdDoc "the terminusdb server");
+
+      package = mkOption {
+        default = pkgs.terminusdb;
+        defaultText = literalExpression "pkgs.terminusdb";
+        description = lib.mdDoc "terminusdb derivation to use.";
+        type = types.package;
+      };
+
+      port = mkOption {
+        default = 6363;
+        defaultText = literalExpression "6363";
+        description = lib.mdDoc "port for terminusdb server.";
+        type = types.port;
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.terminusdb = {
+      description = "Terminusdb is an open-source, document-graph database";
+      documentation = [ "https://terminusdb.com/docs" ];
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      environment = {
+        #https://terminusdb.com/docs/guides/interface-guides/cli#environment
+        TERMINUSDB_SERVER_NAME = "0.0.0.0";
+        TERMINUSDB_SERVER_PORT = "${toString cfg.port}";
+      };
+      serviceConfig = {
```

Comment:
You can use `DynamicUser = true` to let systemd give the service its system user. So you don't need to care about the user creation. In addition, you can use the following hardening parameters to go down to a score of 1.1:
```nix
SystemCallFilter = [
  "@system-service"
  "~@privileged"
  "~@resources"
];
MemoryDenyWriteExecute = true;
RestrictRealtime = true;
RestrictSUIDSGID = true;
ProtectHostname = true;
ProtectKernelModules = true;
ProtectKernelTunables = true;
RestrictNamespaces = true;
NoNewPrivileges = true;
ProtectControlGroups = true;
SystemCallArchitectures = "native";
PrivateTmp = true;
LockPersonality = true;
ProtectSystem = true;
PrivateUsers = true;
ProtectProc = "invisible";
RestrictAddressFamilies = [
  "AF_INET"
  "AF_INET6"
];
ProtectHome = true;
ProtectKernelLogs = true;
PrivateDevices = true;
ProtectClock = true;
DevicePolicy = "closed";
ProcSubset = "pid";
```